src/components/checkbox/checkbox-panel/checkbox-panel.ts

```diff
@@ -0,0 +1,175 @@
+import '../../visual-checkbox/index.js';
+
+/**
+ * Describe the purpose of the component with a single short sentence.
+ *
+ * @slot - Use the unnamed slot to add content to the `sbb-checkbox`.
+ * @slot subtext - Slot used to render a subtext under the label (only visible within a selection panel).
+ * @slot suffix - Slot used to render additional content after the label (only visible within a selection panel).
+ * @event {CustomEvent<void>} didChange - Deprecated. used for React. Will probably be removed once React 19 is available.
+ * @event {Event} change - Event fired on change.
+ * @event {InputEvent} input - Event fired on input.
+ */
+@customElement('sbb-checkbox-panel')
+export class SbbCheckboxPanelElement extends SbbCheckboxCommonElementMixin(
+  SbbUpdateSchedulerMixin(LitElement),
+) {
+  public static override styles: CSSResultGroup = [commonStyle, checkboxPanelStyle];
+  public static readonly events = {
+    didChange: 'didChange',
+    stateChange: 'stateChange',
+    checkboxLoaded: 'checkboxLoaded',
+  } as const;
+
+  /**
+   * Whether the input is the main input of a selection panel.
+   * @internal
+   */
+  public get isSelectionPanelInput(): boolean {
+    return this.hasAttribute('data-is-selection-panel-input');
+  }
+
+  /** The label describing whether the selection panel is expanded (for screen readers only). */
+  @state() private _selectionPanelExpandedLabel!: string;
+
+  private _selectionPanelElement: SbbSelectionExpansionPanelElement | null = null;
+  private _language = new SbbLanguageController(this);
+
+  /**
+   * @internal
+   * Internal event that emits whenever the state of the checkbox
+   * in relation to the parent selection panel changes.
+   */
+  private _stateChange: EventEmitter<SbbCheckboxStateChange> = new EventEmitter(
+    this,
+    SbbCheckboxPanelElement.events.stateChange,
+    { bubbles: true },
+  );
+
+  /**
+   * @internal
+   * Internal event that emits when the checkbox is loaded.
+   */
+  private _checkboxLoaded: EventEmitter<void> = new EventEmitter(
+    this,
+    SbbCheckboxPanelElement.events.checkboxLoaded,
+    { bubbles: true },
+  );
+
+  public constructor() {
+    super();
+    new SbbSlotStateController(this);
+  }
+
+  public override connectedCallback(): void {
+    super.connectedCallback();
+
+    // We can use closest here, as we expect the parent sbb-selection-expansion-panel to be in light DOM.
+    this._selectionPanelElement = this.closest?.('sbb-selection-expansion-panel');
+    this.toggleAttribute('data-is-inside-selection-panel', !!this._selectionPanelElement);
+    this.toggleAttribute(
+      'data-is-selection-panel-input',
+      !!this._selectionPanelElement &&
+        !this.closest?.('sbb-selection-expansion-panel [slot="content"]'),
+    );
+
+    this._checkboxLoaded.emit();
+
+    // We need to call requestUpdate to update the reflected attributes
+    ['disabled', 'required', 'size'].forEach((p) => this.requestUpdate(p));
```

Comment:
1. Does the panel has a `size`?
2. this line is duplicated between here and the checkbox